### PR TITLE
fix(bug): User registration fails

### PR DESCRIPTION
**Pull request by DaspawnW**
Closes https://github.com/gravitee-io/issues/issues/1209

### CODE REVIEW

**Comment by aelamrani** [comment]
Hi @DaspawnW 

Thanks for this PR. Before to check and accept it, we need you to sign our CLA. Could you please refer to https://github.com/gravitee-io/gravitee-management-rest-api/blob/master/CONTRIBUTING.adoc#contributor-license-agreement

Azize

**Comment by brasseld** [comment]
Hi @DaspawnW do not forget to complete CLA if you want to see the fix included in next release 1.16.

Cheers !

File: gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -262,7 +261,14 @@ public UserEntity create(final RegisterUserEntity registerUserEntity) {
 
             // Set date fields
             user.setUpdatedAt(new Date());
-            user = userRepository.update(user);
+
+            if (checkUser.isPresent()) {
```
Comment (aelamrani) [inline gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/UserServiceImpl.java]: Thanks for this PR and sorry for the delay... 
Actually, here the user is confirming his previous registration so we can consider that is existing yet and we should not allow here to create a user without the email validation process. So the update should be enough here. Could you please make the change?

Comment (DaspawnW) [reply gravitee-management-api-service/src/main/java/io/gravitee/management/service/impl/UserServiceImpl.java]: yes sure!

**Comment by aelamrani** [comment]
👍

**Pull request merged**